homeassistant/components/knx/climate.py

```diff
@@ -235,39 +240,74 @@ def max_temp(self):
         await self.async_update_ha_state()
 
     @property
-    def hvac_mode(self):
+    def hvac_mode(self) -> Optional[str]:
         """Return current operation ie. heat, cool, idle."""
+        if self.device.supports_on_off and not self.device.is_on:
+            return HVAC_MODE_OFF
+        elif self.device.supports_on_off and self.device.is_on:
+            return HVAC_MODE_HEAT
         if self.device.mode.supports_operation_mode:
-            return OPERATION_MODES.get(self.device.mode.operation_mode.value)
+            return OPERATION_MODES.get(
+                self.device.mode.operation_mode.value, HVAC_MODE_HEAT)
         return None
 
     @property
-    def hvac_modes(self):
+    def hvac_modes(self) -> Optional[List[str]]:
         """Return the list of available operation modes."""
-        return [OPERATION_MODES.get(operation_mode.value) for
-                operation_mode in
-                self.device.mode.operation_modes]
+        _operations = [OPERATION_MODES.get(operation_mode.value) for
+                       operation_mode in
+                       self.device.mode.operation_modes]
+
+        if self.device.supports_on_off:
+            _operations[HVAC_MODE_HEAT] = HVAC_MODE_HEAT
+            _operations[HVAC_MODE_OFF] = HVAC_MODE_OFF
+
+        return list(filter(None, _operations))
```

Comment:
We prefer list comprehension without using filter.
```py
return [op for op in _operations if op is not None]
```

Reply:
fixed in https://github.com/home-assistant/home-assistant/pull/24942/files